adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -1283,6 +1361,16 @@ func getCorsPolicy(corsConfig *model.CorsConfig) *routev3.CorsPolicy {
 func genRouteCreateParams(swagger *model.MgwSwagger, resource *model.Resource, vHost, endpointBasePath string,
 	prodClusterName string, sandClusterName string, requestInterceptor map[string]model.InterceptEndpoint,
 	responseInterceptor map[string]model.InterceptEndpoint, organizationID string) *routeCreateParams {
+
+	var rateLimitPolicyName, rlMethodDescriptorValue string
+	if swagger.RateLimitLevel == "API" {
+		for _, operation := range resource.GetMethod() {
+			// ALL denotes API level rate limiting
+			rlMethodDescriptorValue = "ALL"
+			rateLimitPolicyName = operation.RateLimitPolicy
```

Comment:
With the new change, we can get the policy name from the API level field.

Reply:
fixed with: https://github.com/wso2/product-microgateway/pull/3142/commits/e7b0b463cf99f802cce280947fbeb86c400927a9